### ISSUE TITLE
[Bug] Issue3408 Failure on Appcenter 4.8

### ISSUE THREAD

**Issue by PureWeen** [xamarin/Xamarin.Forms#12022]
### Description

Appcenter test Issue3408  is crashing 

https://appcenter.ms/orgs/xtc-Xamarin-Forms/apps/XamControl/test/series/pull-12013/runs/ffa2202a-f794-4d4f-a7d4-073af173afd4/96-0-0/39861f62-88c9-4585-9b0f-c34f533ac88c

@rmarinho wasn't able to reproduce locally. 

Running locally using Released mode and setting CI to True inside Environments.props might help reproduce locally

Here's the symbolicated crash

Here's the line that crashes

https://github.com/xamarin/Xamarin.Forms/blob/6d3c41fb2dc0911979cbddda4987e8f76ef7845f/Xamarin.Forms.Platform.iOS/Renderers/ListViewRenderer.cs#L955

```
   XamarinFormsControlGalleryiOS 	0x000000010392edfc xamarin_printf + 50048508 (runtime.m:2472)
5   XamarinFormsControlGalleryiOS 	0x00000001037d11d8 mono_invoke_unhandled_exception_hook + 48615896 (exception.c:1299)
6   XamarinFormsControlGalleryiOS 	0x0000000103766038 mono_handle_exception_internal + 48177208 (mini-exceptions.c:2783)
7   XamarinFormsControlGalleryiOS 	0x0000000103764a20 mono_handle_exception + 48171552 (mini-exceptions.c:3107)
8   XamarinFormsControlGalleryiOS 	0x000000010375ad24 handle_signal_exception + 48131364 (exceptions-arm64.c:526)
9   XamarinFormsControlGalleryiOS 	0x0000000101faafec Xamarin_Forms_Platform_iOS_ListViewRenderer_UnevenListViewDataSource_CalculateHeightForCell_UIKit_UITableView_Xamarin_Forms_Cell + 23293932 (ListViewRenderer.cs:955)
10  XamarinFormsControlGalleryiOS 	0x0000000101faafec Xamarin_Forms_Platform_iOS_ListViewRenderer_UnevenListViewDataSource_CalculateHeightForCell_UIKit_UITableView_Xamarin_Forms_Cell + 23293932 (ListViewRenderer.cs:955)
11  XamarinFormsControlGalleryiOS 	0x0000000101fa9dac Xamarin_Forms_Platform_iOS_ListViewRenderer_UnevenListViewDataSource_GetEstimatedRowHeight_UIKit_UITableView + 23289260 (ListViewRenderer.cs:875)
12  XamarinFormsControlGalleryiOS 	0x0000000101fa9f9c Xamarin_Forms_Platform_iOS_ListViewRenderer_UnevenListViewDataSource_UpdateEstimatedRowHeight_UIKit_UITableView + 23289756 (ListViewRenderer.cs:886)
13  XamarinFormsControlGalleryiOS 	0x0000000101faff54 Xamarin_Forms_Platform_iOS_ListViewRenderer_ListViewDataSource_DetermineEstimatedRowHeight + 23314260 (ListViewRenderer.cs:1338)
14  XamarinFormsControlGalleryiOS 	0x0000000101fb5a8c Xamarin_Forms_Platform_iOS_FormsUITableViewController_ViewWillLayoutSubviews + 23337612 (ListViewRenderer.cs:1681)
15  XamarinFormsControlGalleryiOS 	0x0000000100e3d580 wrapper_runtime_invoke_object_runtime_invoke_dynamic_intptr_intptr_intptr_intptr + 272
16  XamarinFormsControlGalleryiOS 	0x0000000103777c38 mono_jit_runtime_invoke + 48249912 (mini-runtime.c:3165)
17  XamarinFormsControlGalleryiOS 	0x00000001038332ec mono_runtime_invoke_checked + 49017580 (object.c:3220)
18  XamarinFormsControlGalleryiOS 	0x0000000103836b50 mono_runtime_invoke + 49032016 (object.c:3107)
19  XamarinFormsControlGalleryiOS 	0x0000000100a2ea74 native_to_managed_trampoline_14(objc_object*, objc_selector*, _MonoMethod**, unsigned int) + 764532 (registrar.m:519)
20  XamarinFormsControlGalleryiOS 	0x0000000100a550a4 -[Xamarin_Forms_Platform_iOS_FormsUITableViewController viewWillLayoutSubviews] + 921764 (registrar.m:16144)
21  UIKitCore                     	0x00000001e02e3030 0x1df58f000 + 13975600
22  QuartzCore                    	0x00000001b792bc60 0x1b77ed000 + 1305696
23  QuartzCore                    	0x00000001b7930c08 0x1b77ed000 + 1326088
24  QuartzCore                    	0x00000001b78933e4 0x1b77ed000 + 680932
25  QuartzCore                    	0x00000001b78c1620 0x1b77ed000 + 869920
26  QuartzCore                    	0x00000001b78c215c 0x1b77ed000 + 872796
27  CoreFoundation                	0x00000001b337fd08 0x1b32d6000 + 695560
28  CoreFoundation                	0x00000001b337aa30 0x1b32d6000 + 674352
29  CoreFoundation                	0x00000001b337afac 0x1b32d6000 + 675756
30  CoreFoundation                	0x00000001b337a7c0 0x1b32d6000 + 673728
31  GraphicsServices              	0x00000001b557b79c 0x1b5571000 + 42908
32  UIKitCore                     	0x00000001dfe4bc38 0x1df58f000 + 9161784
33  XamarinFormsControlGalleryiOS 	0x0000000101dbc738 wrapper_managed_to_native_UIKit_UIApplication_UIApplicationMain_int_string___intptr_intptr + 344
34  XamarinFormsControlGalleryiOS 	0x0000000101d0eaf0 UIKit_UIApplication_Main_string___intptr_intptr + 20556528 (UIApplication.cs:86)
35  XamarinFormsControlGalleryiOS 	0x0000000101d0e9b4 UIKit_UIApplication_Main_string___string_string + 20556212 (UIApplication.cs:66)
36  XamarinFormsControlGalleryiOS 	0x0000000101d0eab0 UIKit_UIApplication_Main_string___System_Type_System_Type + 20556464 (UIApplication.cs:76)
37  XamarinFormsControlGalleryiOS 	0x0000000100a73234 Xamarin_Forms_ControlGallery_iOS_Application_Main_string__ + 1045044 (Main.cs:9)
38  XamarinFormsControlGalleryiOS 	0x0000000100e3d580 wrapper_runtime_invoke_object_runtime_invoke_dynamic_intptr_intptr_intptr_intptr + 272
39  XamarinFormsControlGalleryiOS 	0x0000000103777c38 mono_jit_runtime_invoke + 48249912 (mini-runtime.c:3165)
40  XamarinFormsControlGalleryiOS 	0x00000001038332ec mono_runtime_invoke_checked + 49017580 (object.c:3220)
41  XamarinFormsControlGalleryiOS 	0x0000000103839544 mono_runtime_exec_main_checked + 49042756 (object.c:5284)
42  XamarinFormsControlGalleryiOS 	0x00000001037561f0 mono_jit_exec + 48112112 (driver.c:1328)
43  XamarinFormsControlGalleryiOS 	0x0000000103941614 xamarin_main + 50124308 (monotouch-main.m:485)
44  XamarinFormsControlGalleryiOS 	0x0000000100a6d38c main + 1020812 (main.m:120)
45  libdyld.dylib                 	0x00000001b2e3e8e0 0x1b2e3d000 + 6368
```

### Reproduction

Regression most likely caused by these changes
https://github.com/xamarin/Xamarin.Forms/commit/975577e837983d5fe656e2bf9caed893c42ffe97#diff-629e73745c456d7bcf063b0fb1290d62R18

- run this `dotnet tool install --global Cake.Tool --version 0.38.4`
- checkout fix_12022  branch
- dotnet cake --target=cg-ios-run-tests --NUNIT_TEST_WHERE="method == Issue3408Test"
- test will fail
- Now you should be able to run CG from vsmac and see the issue if VSMAC doesn't reproduce then run the dotnet cake script again. Make sure to deploy to ios 12 iPhone7
- navigate to test 3408
- With branch `fix_12022` I've fixed the crash but the UI test still fails. If you navigate to 3408 and then click the first button it goes to a blank listview. You should see three items in the listview

The root of the problem is that ListView now has LogicalChildren. The reason you have to run the cake scripts is that that sets up the app so it's being monitored by device agent. The being monitored by Device Agent is what causes the behavior to change on ios 12

**Comment by rmarinho** [comment]
Ok i tried again still can't reproduce.

**Comment by samhouts** [comment]
closed by #12057
